ProjectProtocol/project-protocol-web
# [PRO-193] Handle various user states on "Rate agent" button

🛠️ Changes
---
* Adds RateAgentButton which changes its label and behavior based on 3 states:
  * user logged in + confirmed (can rate)
  * user logged in + not confirmed (show confirmation modal)
  * user not logged in (show sign up and login options)
* Refactors login modal and `openLogin` function to be served by a new LoginUIProvider and `useLogin` hook.
* Adds a ConfirmationModal which alerts the user that they received a confirmation email, and shows a link to resend it.
* Component tests
* Established pattern for checking that the right locale string key is being called
* Added global react-i18n mock.
* A number of en/es locale strings added.

🧪 To Test:
---
* Visit the preview link
* View an agent page without logging in - see login and sign up links
* View an agent page after logging in with an unconfirmed user (try using `unconfirmed@user.com/P@ssword`), try to rate agent and see the confirmation modal.
* Repeat the above as a signed-in, confirmed user and see that you can access the rating form.
